### ISSUE TITLE
Linking fails on Ubuntu 19.04 (g++ 8.3.0) due to missing -pthread link flag

### ISSUE THREAD

**Issue by jkatzmewing** [calref/cboe#194]
Without adding -pthread to posix LINKFLAGS:

```
/usr/bin/ld: build/obj/game/boe.main.o: undefined reference to symbol 'pthread_condattr_setclock@@GLIBC_2.3.3'
/usr/bin/ld: /lib/x86_64-linux-gnu/libpthread.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
scons: *** [build/bin/Blades of Exile] Error 1
scons: building terminated because of errors.
```

With -pthread it links.

Not sure how/if this impacts BSD or Solaris systems (assuming that supporting those is a concern).

**Comment by clort81** [comment]
Did you get problems compiling src/tools/enum_map.hpp on ubuntu?  That gcc did you use?

**Comment by CelticMinstrel** [comment]
Fixed in 45bbc12208f9dd9b35a19edc556f3d37a18aeda7
